### ISSUE TITLE
No support for AccessControls/Clone

### ISSUE THREAD

**Issue by juanreyesv** [citrix/ShareFile-NET#19]
Hi, 

Firstly, thanks for writing this SDK, I have been using it on production for over a year without issues. I'm currently working on adding more features to my application but there seems to be no support for the AccessControls Clone method:
https://api.sharefile.com/rest/docs/resource.aspx?name=AccessControls

Maybe I'm missing something or if I'm right is this planned to be added on a later version? 

Thanks!

**Comment by rgmills** [comment]
Hi @juanreyesv ! I just verified with the latest version on NuGet (https://www.nuget.org/packages/ShareFile.Api.Client/) version 3.3.1476 and it should have the support. Let me know if you are having any trouble with the latest version.

```
var sfClient = new ShareFileClient(baseUrl);
var cloneParams = new AccessControlsCloneParams();
cloneParams.FolderId = "fo123456";
cloneParams.PrincipalId = "sourceUserid";
cloneParams.ClonePrincipalIds = new string[] { "userId" };
var cloneResult = await sfClient.AccessControls.Clone(cloneParams).ExecuteAsync();
```

**Comment by juanreyesv** [comment]
Hi @rgmills , thanks for replying! Ok I will check the NuGet package. Is this repo going to be updated with what's on NuGet? I found that is a bit more flexible sometimes if I can modify the code instead of just consuming a NuGet package.

For instance, in the past I had to create a method to download files and return a byte array rather than the original one which writes the file into a Stream. 

I added two methods to Core/Transfers/Downloaders/AsyncFileDownloader.cs:

`		public async Task<byte[]> DownloadToStreamAsync(CancellationToken? cancellationToken = null,
			Dictionary<string, object> transferMetadata = null)
		{
			var downloadQuery = CreateDownloadQuery();
			var totalBytesToDownload = Item.FileSizeBytes.GetValueOrDefault();
			var progress = new TransferProgress(totalBytesToDownload, transferMetadata);

			using (var stream = await downloadQuery.ExecuteAsync(cancellationToken))
			{
				if (stream != null)
				{
					byte[] buffer = ReadFully(stream);
					NotifyProgress(progress.MarkComplete());
					return buffer;
				}
			}

			NotifyProgress(progress.MarkComplete());
			return null;
		}


		public static byte[] ReadFully(Stream input)
		{
			byte[] buffer = new byte[16 * 1024];
			using (MemoryStream ms = new MemoryStream())
			{
				int read;
				while ((read = input.Read(buffer, 0, buffer.Length)) > 0)
				{
					ms.Write(buffer, 0, read);
				}
				return ms.ToArray();
			}
		}` 

I'm afraid if I use the NuGet package then I will lose this functionality. I will probably remove the use of this methods in the future in favor of downloading the files directly from ShareFile using the API to get the download link.

Thanks again

**Comment by rgmills** [comment]
@juanreyesv Yep, I've been meaning to do it for a while.  I've just merged in #20 which will update the source to represent the latest available from the API.

**Comment by juanreyesv** [comment]
Cool! Thanks heaps for that @rgmills
